databricks_cli/stack/api.py

```diff
@@ -0,0 +1,354 @@
+        :param stack_status: Must have the fields of
+        :return:
+        """
+        self._validate_config(stack_config)
+        if stack_status:
+            self._validate_status(stack_status)
+            resource_id_to_status = self._get_resource_to_status_map(stack_status)
+        else:
+            resource_id_to_status = {}
+
+        stack_name = stack_config[STACK_NAME]
+        click.echo('Deploying stack %s' % stack_name)
+        resource_statuses = []
+
+        for resource_config in stack_config[STACK_RESOURCES]:
+            click.echo()
+            click.echo("Deploying resource")
+            # Retrieve resource deployment info from the last deployment.
+            resource_map_key = (resource_config[RESOURCE_ID], resource_config[RESOURCE_SERVICE])
+            resource_status = resource_id_to_status[resource_map_key] \
+                if resource_map_key in resource_id_to_status else None
+            # Deploy resource, get resource_status
+            new_resource_status = self._deploy_resource(resource_config, resource_status)
+            resource_statuses.append(new_resource_status)
+
+        # stack deploy status is original config with deployed resource statuses added
+        new_stack_status = copy.deepcopy(stack_config)
+        new_stack_status.update({STACK_DEPLOYED: resource_statuses})
+        new_stack_status.update({CLI_VERSION_KEY: CLI_VERSION})
+
+        # Validate that the status has been created correctly
+        self._validate_status(new_stack_status)
+
+        return new_stack_status
+
+    def _deploy_resource(self, resource_config, resource_status=None):  # overwrite to be added
+        """
+        Deploys a resource given a resource information extracted from the stack JSON configuration
+        template.
+
+        :param resource_config: A dict of the resource with fields of RESOURCE_ID, RESOURCE_SERVICE
+        and RESOURCE_PROPERTIES.
+        ex. {'id': 'example-resource', 'service': 'jobs', 'properties': {...}}
+        :param resource_status: A dict of the resource's deployment info from the last
+        deployment. Will be None if this is the first deployment.
+        ex. {'id': 'example-resource', 'service': 'jobs', 'physical_id': {...}}
+        :return: dict resource_status- A dictionary of deployment information of the
+        resource to be stored at deploy time. It includes the resource id of the resource along
+        with the physical id and deploy output of the resource.
+        ex. {'id': 'example-resource', 'service': 'jobs', 'physical_id': {'job_id': 123},
+        'timestamp': 123456789, 'deploy_output': {..}}
+        """
+        resource_id = resource_config[RESOURCE_ID]
+        resource_service = resource_config[RESOURCE_SERVICE]
+        resource_properties = resource_config[RESOURCE_PROPERTIES]
+        physical_id = resource_status[RESOURCE_PHYSICAL_ID] if resource_status else None
```

Comment:
sure, you mean at the call on line 162? https://github.com/databricks/databricks-cli/pull/137/files#diff-adef28b314958b556e396054d0efb708R162

Reply:
I'm going to have `physical_id` and `new_physical_id` instead, since the physical id is semantically the current identifier for the resource before deployment